### PR TITLE
Adds a test version of bioengineering

### LINKED ISSUE #16103: Banana potency isn't affecting slip time.
### What is the problem?
Peels at 100 potency are slipping for incredibly low time, less than a bar of soap.
### Why is it a problem, or what should have happened?
Why even bother with potency if isn't gonna slip?
### How can the problem be reproduced?
Make peels, throw them security officers, watch as they slip,get up, and beat you.
### What revision did the problem happen on? (Helpful, but not required)
39e356c55cf6223162814135ba54512af7481cac

### LINKED ISSUE #16106: Bananna bug
[Everyone]: # (Please fill in your information on the line AFTER the header.)

### What is the problem?
You can not eat the clown banana you spawn with.
### Why is it a problem, or what should have happened?
Not a huge problem its just you can't eat it, every time you try it it says, "There is no more of your banana left!" and then drops it on the ground.
### How can the problem be reproduced?
Just spawn as a clown and try to eat the banana in your backpack.
### What revision did the problem happen on? (Helpful, but not required)
[If you're including this]: # (Type find-server-revision ingame to find it. "Current" isn't helpful.)

[Admins]: # (If you are reporting a bug that occured AFTER you used varedit/admin buttons to alter an object out of normal operating conditions, please verify that you can re-create the bug without the varedit usage/admin buttons before reporting the issue.)

### PR DESCRIPTION
**Pull request by Core0verload**
This PR adds the **Plant DNA Manipulator** machine, one of the biggest parts of my bioengineering system. Thanks to @WJohn for a sprite, I used modified version of his sample scanner machine.

To use it, spawn it, spawn a box of plant gene disks, insert a seed and a disk into the machine and have fun with copypasting genes. Paths for quick and easy spawining:
`/obj/machinery/plantgenes`
`/obj/item/weapon/storage/box/disks_plantgene`

This is still a test version of it, I want to see it spawned by admins and tested by players before making it available to players in every round.

This version lacks any attempts at balancing, and will allow players to copy max stats to every single plant with minimal effort. Balancing is planned in the next PR: it will take a form of failure modes. Each gene insertion/replacement attempt will have a chance to fail, with "Potency 100" gene having only ~5% chance of successful insertion on T1 parts and ~25% on T4. Failure will destroy or mutate the target seed.

Fixes #16103. Fixes #16106.

### CODE REVIEW

**Comment by RandomMarine** [comment]
Anything that offers a better way for getting better plants than constantly alternating between somatoray shots and scanning until the RNG decides to be nice gets my :+1:

**Comment by Iamgoofball** [comment]
>rng chance to fail

:-1: Botany is already RNG enough

:+1: to the rest though

**Comment by RandomMarine** [comment]
To be fair, it's easy to get an absurd excess of identical seeds to run through the manipulator.

**Comment by RandomMarine** [comment]
Modifying content genes doesn't seem to work. In my test case, I removed space drugs from ambrosia vulgaris seeds(Sacrilege, I know.) but they still yielded it when grown.

And attempting to insert a content gene didn't even do anything.

**Comment by Core0verload** [comment]
It has no RNG chances to fail yet. Not going to add these until this gets merged and tested or at least testmerged.

**Comment by RandomMarine** [comment]
>It isn't intended to be applied to every seed. Modify one seed, plant it, harvest it, get a lot of modified seed offsprings.

But you may run through a large amount of seeds until you get a successful modification, is what I mean.

**Comment by Core0verload** [comment]
RNG stuff isn't in yet. I will wait for feedback before adding it. As for reagent genes fix - check it out, i'll upload it in 10 minutes.

**Comment by RandomMarine** [comment]
It was in response to goofball's previous comment... buuuut he edited that out of his post. Joik.

**Comment by Core0verload** [comment]
At least the fix for reagent genes not being reagenty enough is in.

**Comment by JJRcop** [comment]
I like my tomatoes gmo free thank you very much

**Comment by Lumipharon** [comment]
>fill every plant with 100 potency worth of lexorin/whatever novaflowers have
>all of the chefs food sets you on fire and suffocates you in 2 seconds flat

Very neat though, always good to see botany improvements.

File: code/modules/hydroponics/gene_modder.dm [hunk 1]
```diff
@@ -0,0 +1,398 @@
...
+		icon_state = "dnamod-off"
+	else
+		icon_state = "dnamod"
+	if(seed)
+		overlays += "dnamod-dna"
+	if(panel_open)
+		overlays += "dnamod-open"
+
+/obj/machinery/plantgenes/attackby(obj/item/I, mob/user, params)
+	if(default_deconstruction_screwdriver(user, "dnamod", "dnamod", I))
```
Comment (Razharas) [inline code/modules/hydroponics/gene_modder.dm]: This handles icon chaning on its own, thats why you pass 2 arguments with icon states, one for open one for close

Comment (Core0verload) [inline code/modules/hydroponics/gene_modder.dm]: It can't handle my overlays. And it uses overlays because it has a depowered icon_state.

**Comment by FalseIncarnate** [comment]
Out of curiosity, what are the major differences between this and the baycode xenobotany machines?

**Comment by Core0verload** [comment]
Amount of autism is greatly reduced. No DNA degradation, no discovery, and all the genes are separated. You can just add Space Drugs gene extracted from Ambrosia to apple and grow drugged apples. Or remove Amatoxin from fly amanitas to safely dry and smoke them.

**Comment by FalseIncarnate** [comment]
Ok, so it seems that you have the ability to affect individual chems and traits as well as just merge together traits, which the baycode version of this doesn't quite allow for currently. Their stuff essentially has sets of traits that get packaged together, so things like the yield stat are bundled with other stats making it more time-consuming to really fine-tune your creations. They also are "all-or-nothing" on the chems, so you can't just splice one chemical out of the three onto a new plant without also splicing on the others.

Thanks for quick and dirty summary of differences, I'll definitely be keeping an eye on this system's development to see what could be used to improve Paradise's botany system more (already removed the annoying gene-masking that Bay used).

**Comment by Lumipharon** [comment]
So... is this getting merged any time soon?

**Comment by Core0verload** [comment]
Who knows?

**Comment by Lumipharon** [comment]
ayyy

**Pull request merged**